runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/testing/TestDataflowRunner.java

```diff
@@ -174,24 +182,22 @@ DataflowPipelineJob run(Pipeline pipeline, DataflowRunner runner) {
     return runner.apply(transform, input);
   }
 
-  Optional<Boolean> checkForSuccess(DataflowPipelineJob job)
+  Optional<Boolean> checkForSuccess(DataflowPipelineJob job, JobMetrics metrics)
```

Comment:
Add javadoc? explain "success" and its return values.
Maybe renamed to passedPAssertCheck?

And, what will be returned for the following four streaming cases:
1. streaming PAssert tests failed
passedPAssertCheck: false
atMaxWatermark: could be true or false, depends?
2. streaming PAssert tests succeeded
passedPAssertCheck: true
atMaxWatermark: true
3. streaming non-PAssert tests failed
passedPAssertCheck: absent
atMaxWatermark: false
4. streaming non-PAssert tests succeeded
passedPAssertCheck: absent
atMaxWatermark: true

What they would return for batch cases?

Could lines 122-128 be simplified?